blocks/library/freeform/format-list.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import clickOutside from 'react-click-outside';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, Dashicon } from 'components';
+
+/**
+ * Internal dependencies
+ */
+import './format-list.scss';
+
+class FormatList extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.switchFormat = this.switchFormat.bind( this );
+		this.toggleMenu = this.toggleMenu.bind( this );
+		this.state = {
+			open: false,
+		};
+	}
+
+	handleClickOutside() {
+		if ( ! this.state.open ) {
+			return;
+		}
+		this.setState( { open: false } );
+	}
+
+	toggleMenu() {
+		this.setState( {
+			open: ! this.state.open,
+		} );
+	}
+
+	switchFormat( newValue ) {
+		if ( this.props.value !== newValue && this.props.onFormatChange ) {
+			this.props.onFormatChange( newValue );
+		}
+		this.setState( { open: false } );
+	}
+
+	render() {
+		const { formats = [ { text: '', value: '', textStyle: () => '' } ] } = this.props;
```

Comment:
Probably not. 
I was thinking that there would be a render step when the editor hadn't loaded and there would be no formats. I can probably handle this cleaner though so I will revise it.